zipline/data/us_equity_pricing.py

```diff
@@ -767,35 +767,38 @@ def __init__(self, calendar, panel):
             # Fake volume if it does not exist.
             panel.loc[:, :, 'volume'] = int(1e9)
 
-        self.first_trading_day = panel.major_axis[0]
-        self._calendar = calendar
+        self.first_trading_day = normalize_date(panel.major_axis[0])
```

Comment:
Yea, I assume the caller constructed the list of dts from a trading calendar, so probably they could pass the whole thing along, and we can make use of more of its methods here.  (So whatever the name of the arg, yes, I agree we should pass a trading calendar instance instead of just its sessions.)